sqlalchemy_bigquery/base.py

```diff
@@ -56,7 +56,10 @@
 from .parse_url import parse_url
 from . import _helpers, _struct, _types
 
-FIELD_ILLEGAL_CHARACTERS = re.compile(r"[^\w]+")
+# Illegal characters is intended to be all characters that are not explicitly
+# allowed as part of the flexible column names.
+# https://cloud.google.com/bigquery/docs/schemas#flexible-column-names
+FIELD_ILLEGAL_CHARACTERS = re.compile(r"[^\w&%=+:'<>#|\n]+", re.ASCII)
```

Comment:
Let's just list out the characters that aren't allowed in https://cloud.google.com/bigquery/docs/schemas#flexible-column-names

```suggestion
FIELD_ILLEGAL_CHARACTERS = re.compile(r"[!"$()*,./;?@[\\\]^{}~\n]+", re.ASCII)
```